bigcache.go

```diff
@@ -33,6 +33,121 @@ type cacheShard struct {
 	onRemove    func(wrappedEntry []byte)
 }
 
+// EntryInfo holds informations about entry in the cache
+type EntryInfo struct {
+	entry []byte
+}
+
+// Key returns entry's underlying key
+func (e EntryInfo) Key() string {
+	return readKeyFromEntry(e.entry)
+}
+
+// Hash returns entry's hash value
+func (e EntryInfo) Hash() uint64 {
+	return readHashFromEntry(e.entry)
+}
+
+// Timestamp returns entry's timestamp (time of insertion)
+func (e EntryInfo) Timestamp() uint64 {
+	return readTimestampFromEntry(e.entry)
+}
+
+// Value returns entry's underlying value
+func (e EntryInfo) Value() []byte {
+	return readEntry(e.entry)
+}
+
+// EntryInfoIterator allows to iterate over entries in the cache
+type EntryInfoIterator struct {
+	sync.Mutex
+	cache        *BigCache
+	currentShard int
+	currentIndex int
+	elements     []uint32
+	valid        bool
+}
+
+func copyCurrentShardMap(shard *cacheShard) []uint32 {
+	shard.lock.RLock()
+	defer shard.lock.RUnlock()
+
+	var elements []uint32
+
+	for _, index := range shard.hashmap {
+		elements = append(elements, index)
+	}
+
+	return elements
+}
+
+// HasNext returns true if there is next element in the iterator, false otherwise
+func (it *EntryInfoIterator) HasNext() bool {
+	it.Lock()
+	defer it.Unlock()
+
+	it.valid = false
+	it.currentIndex++
+
+	if !(len(it.elements) > it.currentIndex) {
```

Comment:
Oh, I was thinking about early return `true` so the code below doesn't need to be indent.

```
if len(it.elements) > it.currentIndex {
   it.valid = true;
    return true
}
```